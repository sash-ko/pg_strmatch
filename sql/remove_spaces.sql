CREATE OR REPLACE FUNCTION remove_spaces(text) RETURNS text
    AS 'pg_strmatch.so', 'remove_spaces'
    LANGUAGE C STRICT;

SELECT remove_spaces('') = '' AS;
SELECT remove_spaces(NULL) IS NULL;
SELECT remove_spaces('   ') = '';
SELECT remove_spaces('no  space') = 'no space';
SELECT remove_spaces(' spaces  ') = 'spaces';
SELECT remove_spaces('  more  and more spaces!!  ') = 'more and more spaces!!';
SELECT remove_spaces('no_space') = 'no_space';
