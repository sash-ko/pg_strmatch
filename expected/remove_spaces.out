CREATE OR REPLACE FUNCTION remove_spaces(text) RETURNS text
    AS 'pg_strmatch.so', 'remove_spaces'
    LANGUAGE C STRICT;
SELECT remove_spaces('') = '' AS;
ERROR:  syntax error at or near ";"
LINE 1: SELECT remove_spaces('') = '' AS;
                                        ^
SELECT remove_spaces(NULL) IS NULL;
 ?column? 
----------
 t
(1 row)

SELECT remove_spaces('   ') = '';
 ?column? 
----------
 t
(1 row)

SELECT remove_spaces('no  space') = 'no space';
 ?column? 
----------
 t
(1 row)

SELECT remove_spaces(' spaces  ') = 'spaces';
 ?column? 
----------
 t
(1 row)

SELECT remove_spaces('  more  and more spaces!!  ') = 'more and more spaces!!';
 ?column? 
----------
 t
(1 row)

SELECT remove_spaces('no_space') = 'no_space';
 ?column? 
----------
 t
(1 row)

